Added audio, play/pause button and audio switch, image and audio file…

zseiman opened the pull request:
…s as well as corresponding directories

HTML
=>div #mute
=>audio
css
=>mute
=>mute:hover
directories
=>images
=>music
JavaScript
=>MusicSwitch function and associated commands
images
=>muteButton.png
=>mutedButton.png
audio
=>bensound-scifi.mp3
text
=>Credits (url to music source)

Purpose: to add background music to game along with optional on/off switch

zseiman committed:
Added audio, play/pause button and audio switch, image and audio files as well as corresponding directories

HTML
=>div #mute
=>audio
css
=>mute
=>mute:hover
directories
=>images
=>music
JavaScript
=>MusicSwitch function and associated commands
images
=>muteButton.png
=>mutedButton.png
audio
=>bensound-scifi.mp3
text
=>Credits (url to music source)

aapomm commented:
@zseiman thank you for this pull request! I appreciate it. Though, there are some concerns that need to be addressed before I can merge this.

aapomm commented on css/main.css:
```diff
@@ -28,3 +28,13 @@ body {
 
   text-align: right;
 }
+#mute{
```
@zseiman can you move this to `utility.css`? This is to preserve the files' semantics. I think the mute button is a "utility" and shouldn't belong in `main.css`. Thanks.

Also, the mute button seems to push the whole console off center. I think it would be better if the button is placed at the top left/right preferably with `position: absolute`.

aapomm commented on music/Credits.txt:
```diff
@@ -0,0 +1 @@
+Music source:http://www.bensound.com/royalty-free-music/track/sci-fi
```
@zseiman You can move this to README.md instead.

aapomm commented on js/main.js:
```diff
@@ -1,3 +1,8 @@
+// Start Music
```
@zseiman I suggest you create a separate js file solely for music. Call it `music.js` and include it in `index.html`. You can move all music related js there enclosed under a `$(function(){});` block (see `command.js').

aapomm commented on index.html:
```diff
@@ -32,6 +32,11 @@
       </div>
     </div>
+    <div id="mute">
+      <audio id="sound">
```
@zseiman add the attribute `autoplay` and `loop`.

aapomm commented on js/main.js:
```diff
@@ -21,6 +26,20 @@ function Main(){
+}
+document.getElementById('mute').addEventListener('click', MusicSwitch);
+//restarts music once track has ended
+document.getElementById('sound').addEventListener('onended', audio.play());
```
@zseiman you wont need this once you add the `loop` attribute to the `<audio>` tag.

aapomm commented on js/main.js:
```diff
@@ -1,3 +1,8 @@
+// Start Music
+var audio = document.getElementById('sound');
+var sw = true;
+audio.play();
```
@zseiman you won't need this once you add the `autoplay` tag.

aapomm commented on js/main.js:
```diff
@@ -21,6 +26,20 @@ function Main(){
 
+//Function for stopping and starting audio playback -Z
+function MusicSwitch(){
+if(sw){
```
@zseiman instead of using a boolean variable `sw` to check if the music is playing, you can use the property `paused` of a HTML audio tag. So in this case, `if (sw){` will be `if (!audio.paused){`.

aapomm commented on js/main.js:
```diff
@@ -21,6 +26,20 @@ function Main(){
 }
 
+//Function for stopping and starting audio playback -Z
+function MusicSwitch(){
```
@zseiman rename this to `musicSwitch`. This is because in most programming languages a capitalized name implies a class (see `Game` and `Main`).

aapomm commented on css/main.css:
```diff
@@ -28,3 +28,13 @@ body {
+}
+#mute:hover{
+  background-image: url(../images/muteButton.png);
+}
```
@zseiman add a newline.

chuckoy commented:
For linking: fix for issue #10

zseiman commented:
Awesome! Yea, I'm happy to try to help. I'm just stoked to have an opportunity to contribute to something. Thanks for the response!

zseiman committed:
Added background music as well as mute button

=>index.html
added audio tag with autoplay and loop attributes
=> +music.js
added music.js containing 'musicSwitch' function
=>utility.css
added #mute button styling

aapomm commented on css/main.css:
```diff
@@ -27,4 +27,3 @@ body {
   height: 27px;
 
   text-align: right;
-}
```
@zseiman I think you accidentally removed this. haha

zseiman replied:
Whoops! Sorry about that. Have to work on my code cleanliness. This is my first attempt at contribution and I'm pretty nervous about it.

aapomm replied:
Relax. You're doing fine! 👍

aapomm commented on js/music.js:
```diff
@@ -0,0 +1,13 @@
+$(function(){
```
@zseiman can you fix the indentations here? thanks!

zseiman committed:
Small fixes

fixed bracket in main.css and indentation in music.js
